pkgs/development/python-modules/kaleido/default.nix

```diff
@@ -0,0 +1,77 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchurl,
+  autoPatchelfHook,
+  expat,
+  libGL,
+  nspr,
+  nss,
+  sqlite,
+  system ? builtins.currentSystem,
+}:
+
+buildPythonPackage rec {
+  pname = "kaleido";
+  version = "0.2.1.post1";
+  format = "wheel";
+
+  src =
+    let
+      pypiPackageName = "kaleido";
+    in
+    {
+      # This library is so cursed that I have to use fetchurl instead of fetchPypi. I am not happy.
+      x86_64-linux = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl";
+        hash = "sha256-qiHPG/HHj4+lCp99ReEAPDh709b+CnZ8+780S5W9w6g=";
+      };
+      aarch64-linux = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1.post1-py2.py3-none-manylinux2014_armv7l.whl";
+        hash = lib.fakeHash;
+      };
+      x86_64-darwin = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1-py2.py3-none-macosx_10_11_x86_64.whl";
+        hash = lib.fakeHash;
+      };
+      aarch64-darwin = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1-py2.py3-none-macosx_11_0_arm64.whl";
+        hash = lib.fakeHash;
+      };
+    }
+    ."${system}" or (throw "Unsupported system for ${pname}: ${system}");
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [
+    expat
+    libGL
+    nspr
+    nss
+    sqlite
+  ];
+
+  pythonImportsCheck = [ "kaleido" ];
+
+  postInstall = ''
+    mkdir -p $out/bin
+    cp $out/lib/python*/site-packages/kaleido/executable/bin/kaleido $out/bin
+    wrapProgram $out/bin/kaleido \
+      --set FONTCONFIG_PATH $out/lib/python*/site-packages/kaleido/executable/etc/fonts \
+      --set XDG_DATA_HOME $out/lib/python*/site-packages/kaleido/executable/xdg
+  '';
+
+  meta = {
+    description = "Fast static image export for web-based visualization libraries with zero dependencies";
+    homepage = "https://github.com/plotly/Kaleido";
+    changelog = "https://github.com/plotly/Kaleido/releases";
+    platforms = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-linux"
+      "aaarch64-darwin"
+    ];
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ]; # Trust me, I'm not happy. But after literal hours of trying to reverse-engineer their build system and getting nowhere, I'll use the stupid binaries >:(
```

Comment:
This is the third attempt, as far as I know, but no one has been able to build it from the source.😢